backend/de.metas.manufacturing/src/main/java/org/eevolution/process/PP_Product_BOM_Check.java

```diff
@@ -160,15 +162,21 @@ private void checkProductById(@NonNull final I_M_Product product)
 		{
 			final ProductId productId = ProductId.ofRepoId(tbomline.getM_Product_ID());
 			final I_M_Product bomLineProduct = productBL.getById(productId);
-			updateProductLLCAndMarkAsVerified(bomLineProduct);
+			checkProductBOMCyclesAndMarkAsVerified(bomLineProduct);
 		}
 	}
 
-	private void updateProductLLCAndMarkAsVerified(final I_M_Product product)
+	private void checkProductBOMCyclesAndMarkAsVerified(final I_M_Product product)
 	{
-		// NOTE: when LLC is calculated, the BOM cycles are also checked
-		final int lowLevelCode = productBOMBL.calculateProductLowestLevel(ProductId.ofRepoId(product.getM_Product_ID()));
-		product.setLowLevel(lowLevelCode);
+		// Checking BOM cycles
+		try
+		{
+			productBOMBL.createParentProductNode(ProductId.ofRepoId(product.getM_Product_ID()));
+		}
+		catch (final BOMCycleException e)
+		{
+			throw new LiberoException("Cycle detected in BOM for product: " + product.getValue());
+		}
```

Comment:
consider throwing this error from ProductBOMCycleDetection

Reply:
done